machine/v1beta1/types_vsphereprovider.go

```diff
@@ -51,6 +51,11 @@ type VSphereMachineProviderSpec struct {
 	// This parameter will be ignored if 'LinkedClone' CloneMode is set.
 	// +optional
 	DiskGiB int32 `json:"diskGiB,omitempty"`
+	// TagIDs is an optional set of tags to add to an instance. Specified tagIDs
+	// must use URN-notation instead of display names.
+	// For example: urn:vmomi:InventoryServiceTag:5736bf56-49f5-4667-b38c-b97e09dc9578:GLOBAL
+	// +optional
+	TagIDs []string `json:"tagIDs,omitempty"`
```

Comment:
Is there a limit to the number of tags that can be applied?

Reply:
There isn't a hard limit upstream but excessive tags could cause latency in machine reconciliation.  We could impose an arbitrary maximum.

Reply:
There isn't a hard limit, but there is performance hit if we allow too many tags to be reconciled and attached.  For now, setting a 10 tag limit.